lib/motors/controller.cpp

```diff
@@ -1,9 +1,98 @@
 #include "controller.hpp"
 
+#include <fstream>
+
 namespace hyped::motors {
 
-Controller::Controller(core::ILogger &logger) : logger_(logger)
+Controller::Controller(core::ILogger &logger) : logger_(logger), configuration_messages_()
+{
+}
+
+core::Result Controller::parseMessageFile(const std::string &path)
+{
+  std::ifstream input_stream(path);
+  if (!input_stream.is_open()) {
+    logger_.log(core::LogLevel::kFatal, "Failed to open file %s", path.c_str());
+    return core::Result::kFailure;
+  }
+  rapidjson::IStreamWrapper input_stream_wrapper(input_stream);
+  rapidjson::Document document;
+  rapidjson::ParseResult result = document.ParseStream(input_stream_wrapper);
+  if (!result) {
+    logger_.log(core::LogLevel::kFatal,
+                "Error parsing JSON: %s",
+                rapidjson::GetParseError_En(document.GetParseError()));
+    return core::Result::kFailure;
+  }
+  if (!document.HasMember("config_messages")) {
+    logger_.log(core::LogLevel::kFatal,
+                "Missing required field 'config_messages' in can message file at %s",
+                path.c_str());
+    return core::Result::kFailure;
+  }
+  if (!document.HasMember("messages")) {
+    logger_.log(core::LogLevel::kFatal,
+                "Missing required field 'messages' in can message file at %s",
+                path.c_str());
+    return core::Result::kFailure;
+  }
+  const auto configuration_messages = document["config_messages"].GetArray();
+  for (auto &message : configuration_messages) {
+    const std::optional<core::CanFrame> new_message = parseJsonCanFrame(message.GetObject());
+    if (!new_message) {
+      logger_.log(core::LogLevel::kFatal,
+                  "Invalid CAN configuration frame in JSON message file at path %s",
+                  path.c_str());
+      return core::Result::kFailure;
+    }
+    configuration_messages_.push_back(new_message.value());
+  }
+  const auto messages = document["messages"].GetObject();
+  for (auto &message : messages) {
+    const std::optional<core::CanFrame> new_message = parseJsonCanFrame(message.value.GetObject());
+    if (!new_message) {
+      logger_.log(
+        core::LogLevel::kFatal, "Invalid CAN frame in JSON message file at path %s", path.c_str());
+      return core::Result::kFailure;
+    }
+    auto x = new_message.value();
```

Comment:
this line seems a bit pointless? `x` doesn't tell me what it is, and its not constant

juts do `*new_message` below?